### PR TITLE
BeninSmallHolderCashews: return geospatial metadata in sample

### PR DESCRIPTION
**Pull request by recursix**
exposing geoinformation in the return dict of __getitem__ for benin_cashews.py

### CODE REVIEW

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:236]:
```suggestion
            image, mask, transform, crs, and metadata at that index
```

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:251]: Might make more sense to call it "transform" since that's what rasterio calls it.
```suggestion
            "transform": self.tile_transform,
```

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:348]: Instead of setting these attributes, it might be safer to return these from the function.

File: torchgeo/datasets/benin_cashews.py
Comment (recursix) [reply torchgeo/datasets/benin_cashews.py:251]: I went with geotransform since you already have something called transform for the datasets, but I'm ok with it.

File: torchgeo/datasets/benin_cashews.py
Comment (recursix) [reply torchgeo/datasets/benin_cashews.py:348]: I agree, I went with the original style that was used for passing tile_transform to the mask function.

Do you want me to change the whole thing?

Comment (adamjstewart) [reply torchgeo/datasets/benin_cashews.py:348]: Let's see if @calebrob6 has an opinion on this before changing anything, he wrote the original dataset class.

Comment (calebrob6) [reply torchgeo/datasets/benin_cashews.py:348]: returning `crs` and `tile_transform` would be less brittle / better overall, however this way wouldn't break anything

Comment (recursix) [reply torchgeo/datasets/benin_cashews.py]: done

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:238]:
```suggestion
            a dict containing image, mask, tile_transform, crs, and metadata at that index
```

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:253]: @calebrob6 opinions on "transform" vs. "tile_transform" vs. "geotransform" before we standardize this?

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:287]: I'm not sure if black is going to like this. Just run `black .` to solve most formatting issues.

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:299]: In Sphinx, "Args" and "Returns" have very different format. "Args" can have multiple strings, and long lines should be indented to denote that it is a continuation. "Returns" can only have a single string and on indentation should be used (so this was actually broken before, not that it makes it to the generated docs). So I would remove indentation and make this a single paragraph/sentence.

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:333]: Same comment here, need commas to denote that this is a single sentence

**Comment by recursix** [comment]
Have the test run? Many of the are reported as "expected"

I'm still waiting to get approval to sign the CLA.

**Comment by adamjstewart** [comment]
For first time contributors GitHub requires maintainers to explicitly run the tests (just did). This is a security feature to prevent bitcoin mining on CI resources. We could disable this.

**Comment by adamjstewart** [comment]
I wouldn't both using `autopep8`, we use `black` and `isort` instead. See https://torchgeo.readthedocs.io/en/latest/user/contributing.html#linters if you haven't already.

**Comment by recursix** [comment]
> I wouldn't both using `autopep8`, we use `black` and `isort` instead. See https://torchgeo.readthedocs.io/en/latest/user/contributing.html#linters if you haven't already.

I'm trying to avoid unconfiguring my environment and configuring it just for that PR.

**Comment by recursix** [comment]
Ok, I switched to black and installed pre-commit hooks

**Comment by adamjstewart** [comment]
@calebrob6 did we ever get the CLA issue resolved?

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [inline torchgeo/datasets/benin_cashews.py:237]:
```suggestion
            a dict containing image, mask, tile_transform, crs, and metadata at index.
```

File: torchgeo/datasets/benin_cashews.py
Comment (adamjstewart) [reply torchgeo/datasets/benin_cashews.py:252]: @calebrob6 any opinions on this?

**Comment by calebrob6** [comment]
Yes the CLA bot currently works I think

**Comment by recursix** [comment]
I was still waiting on the approval to sign the CLA (manager in vacation).

Thanks.

File: torchgeo/datasets/benin_cashews.py
Comment (calebrob6) [reply torchgeo/datasets/benin_cashews.py:252]: slight preference for "transform" as that is what rasterio calls it

Comment (adamjstewart) [reply torchgeo/datasets/benin_cashews.py:252]: That was my slight preference too

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Can you replace all `tile_transform` with `transform`?

**Comment by recursix** [comment]
will do. I'm about to leave on vacation. I might have to wait after.

**Comment by calebrob6** [comment]
I can do it no problem

**Review by adamjstewart — APPROVED** [review]

**Comment by recursix** [comment]
Awesome!

**Comment by adamjstewart** [comment]
Thanks for the PR @recursix! Hopefully we can get this converted to a `GeoDataset` in the future when all of us have more free time.

**Pull request merged**
